cmd/launcher/main.go

```diff
@@ -154,6 +144,76 @@ func runSubcommands() error {
 
 }
 
+// runNewerLauncherIfAvailable checks the autoupdate library for a newer version
+// of launcher than the currently-running one. If found, it will exec that version.
+func runNewerLauncherIfAvailable(ctx context.Context, logger log.Logger) {
+	// If the legacy autoupdate path variable isn't already set, set it to help
+	// the legacy autoupdater find its update directory even when the newer binary
+	// runs out of a different directory.
+	if _, ok := os.LookupEnv(autoupdate.LegacyAutoupdatePathEnvVar); !ok {
+		currentPath, err := os.Executable()
+		if err == nil {
+			os.Setenv(autoupdate.LegacyAutoupdatePathEnvVar, currentPath)
+		}
+	}
+
+	newerBinary, err := latestLauncherPath(ctx, logger)
+	if err != nil {
+		logutil.Fatal(logger, "msg", "checking for updated version", "err", err)
+	}
+
+	if newerBinary == "" {
+		level.Debug(logger).Log("msg", "nothing newer")
+		return
+	}
+
+	level.Debug(logger).Log(
+		"msg", "preparing to exec new binary",
+		"old_version", version.Version().Version,
+		"new_binary", newerBinary,
+	)
+
+	if err := execwrapper.Exec(ctx, newerBinary, os.Args, os.Environ()); err != nil {
+		logutil.Fatal(logger, "msg", "error execing newer version of launcher", "new_binary", newerBinary, "err", err)
+	}
+
+	logutil.Fatal(logger, "msg", "execing newer version of launcher exited unexpectedly without error", "new_binary", newerBinary)
+}
+
+// latestLauncherPath looks for the latest version of launcher in the new autoupdate library,
+// falling back to the old library if necessary.
+func latestLauncherPath(ctx context.Context, logger log.Logger) (string, error) {
+	newerBinary, err := tuf.CheckOutLatestWithoutConfig("launcher", logger)
+	if err != nil {
+		level.Error(logger).Log(
```

Comment:
Won't this behavior trigger for everyone?

Reply:
Ah, I see the legacy switching is happening down in the tuf library. I don't know what feels less risky. But sure...

Reply:
I tried it both ways, and ultimately it seemed a little better to keep the logic cleaner/simpler in main.go